### ISSUE TITLE
[ambient] Disable L7 istio configs for services that are in Ambient but doesn't have a waypoint proxy

### ISSUE THREAD

**Issue by josunect** [kiali/kiali#7900]
### What do you want to improve?
If a service is in Ambient but just with L4 processing, it shouldn't let the user create L7 traffic policies, as they don't be applicable:
![image](https://github.com/user-attachments/assets/8749e0ee-8057-49c1-898c-196e6913a205)
t might confuse the user if Kiali let them create the istio config but then it is not taking effect.

### What is the current behavior?
If a service is in Ambient with just L4 processing, Kiali enables L7 istio configs.

### What is the new behavior?
If the service is in an Ambient Mesh, but it doesn't have a waypoint, should disable L7 istio configs not applicable. I think just the TCP Traffic shifting should be enabled:

![image](https://github.com/user-attachments/assets/2bda0e0c-0785-4d71-9740-83481a28b339)

Ref. https://github.com/kiali/kiali/issues/7826

**Comment by jshaughn** [comment]
Instead of blocking config creation, I wonder if we should treat this more as a troubleshooting issue, and report validation errors whenever there is L7 config for a component that is not captured by a waypoint?  Because it could be that it was captured at create-time, but later the waypoint was removed.  Also, I suppose a user may want to create L7 config prior to defining the waypoint.

**Comment by josunect** [comment]
> Instead of blocking config creation, I wonder if we should treat this more as a troubleshooting issue, and report validation errors whenever there is L7 config for a component that is not captured by a waypoint? Because it could be that it was captured at create-time, but later the waypoint was removed. Also, I suppose a user may want to create L7 config prior to defining the waypoint.

I think that would be helpful

**Comment by jshaughn** [comment]
@josunect Do you think we should pursue an improvement here?

**Comment by josunect** [comment]
We might want to create a warning or a validation message when an istio configuration acts in L7 but there is no waypoint applying to the service? (For Ambient namespaces). 
If not, anyway, an mcp feature was added here with a similar purpose: https://github.com/kiali/kiali/pull/9989 that can reuse some of the logic, or just have this feature live in the AI side

**Comment by jshaughn** [comment]
I guess the difference is that validations are performed automatically and the MCP would need to be asked to check? Or not?

**Comment by josunect** [comment]
> I guess the difference is that validations are performed automatically and the MCP would need to be asked to check? Or not?

Yes, right. So if we had this logic in the validations, maybe the MCP could make use of this information, I think

**Comment by jshaughn** [comment]
It is not too hard to make a mistake such that a waypoint does not capture an intended service. So, it seems reasonable to add an actual validation that looks for "dead" traffic routing config. Would that just be any virtual service on an uncaptured service?

**Comment by josunect** [comment]
> It is not too hard to make a mistake such that a waypoint does not capture an intended service. So, it seems reasonable to add an actual validation that looks for "dead" traffic routing config. Would that just be any virtual service on an uncaptured service?

I think a validation for "dead" L7 routing config makes sense.

What we already have in MCP: the analyze_ambient_policies tool classifies configs as L4 vs L7 and warns when an L7 config is in an Ambient namespace with no waypoint at all. However, once a namespace has any waypoint, the tool assumes all L7 configs will work, it does not check per-service capture (no use of GetWaypointsForService / capture labels). 

What I think we should add:

- Ambient namespace + L7 config + no waypoint: config won't be enforced (What the MCP is doing ATM)
- namespace has a waypoint, but a specific service isn't captured (missing/wrong istio.io/use-waypoint, or use-waypoint: none). 

For VirtualService specifically I think it should be more complex, perhaps it should need to resolve spec.hosts to a service and validate if they are enrolled in a wapoint?

**Comment by jshaughn** [comment]
> For VirtualService specifically I think it should be more complex, perhaps it should need to resolve spec.hosts to a service and validate if they are enrolled in a waypoint?

Yes, something like that. It may be easier to just identify services that have associated L7 config, and check if they are captured. Anyway, it sounds like this is something we can pursue.

Feel free to remove Refinement label and add the Backlog label, or if you prefer to take near-term action, just assign to the sprint project.

**Comment by josunect** [comment]
### L7 configs in scope for this PR

| Istio config | L7? | Same namespace as destination? | What to validate | Proposed code |
|---|---|---|---|---|
| **AuthorizationPolicy** (methods / paths / headers / JWT) | Yes (when L7 fields are used) | **Yes** — only applies to workloads in the same namespace | Ambient namespace + L7 + no waypoint enrollment (namespace / workload) | KIA0109 |
| **AuthorizationPolicy** (L4 only: principals, ports, namespaces) | No | Yes (same namespace), but **no waypoint required** | No L7 warning | — |
| **RequestAuthentication** | Yes (always) | **Yes** — same namespace as the workloads | Ambient namespace + no enrollment | KIA1110 |
| **PeerAuthentication** | No (L4 / ztunnel) | Yes | Out of L7 scope | — |
| **VirtualService** HTTP/TLS with `gateways: []` or `mesh` | Yes | **Yes in practice** (verified: no effect in `test`; works in `bookinfo`). Aligns with Istio DR lookup path / Ambient destination model | Resolve `hosts` → Service; destination enrollment; prefer CR in the service namespace | KIA1109 / KIA1113 |
| **VirtualService** TCP-only | No (L4) | — | No warning | — |
| **VirtualService** ingress/egress Gateway only | L7 at Gateway, **not** at waypoint | N/A for waypoint | No waypoint warning | — |
| **DestinationRule** L7 (HTTP pool, outlier detection, HTTP consistent-hash) | Yes | **Yes in practice** (same pattern as VS). Official DR lookup: client ns → **service ns** → root ns | Resolve `host` → Service; destination enrollment; prefer CR in the service namespace | KIA0210 / KIA0211 |
| **DestinationRule** TLS / simple LB only | No | — | No warning | — |
| **WasmPlugin** | Yes | **Yes** — selector / destination workload namespace | Ambient namespace + no enrollment (or unmatched workloads not enrolled) | KIA1111 |
| **Telemetry** (tracing, access logging, metric overrides) | Yes | **Yes** (namespace / selector scope) | Ambient namespace + no enrollment | KIA1112 |
| **Telemetry** basic TCP metrics only | No | — | No warning | — |

### Quick rules

- **Security / Wasm / Telemetry** → same namespace as the workload; validate enrollment there.
- **VirtualService / DestinationRule** → destination Service must be enrolled; in Ambient the CR should live in the **service namespace** (cross-namespace did not take effect in our tests; DR lookup path is documented).
- **L4** → enforced by ztunnel; do not require a waypoint.

### References

- [Traffic management best practices (DestinationRule lookup path)](https://istio.io/latest/docs/ops/best-practices/traffic-management/)
- [Configure waypoint proxies](https://istio.io/latest/docs/ambient/usage/waypoint/)
- [Migrate policies (Ambient / VS alpha / `targetRefs`)](https://istio.io/latest/docs/ambient/migrate/migrate-policies/)
